pkg/cmd/infra/deployer/deployer.go

```diff
@@ -196,5 +203,53 @@ func (d *Deployer) Deploy(namespace, deploymentName string) error {
 	} else {
 		glog.Infof("Deploying from %s to %s (replicas: %d)", deployutil.LabelForDeployment(from), deployutil.LabelForDeployment(to), desiredReplicas)
 	}
-	return strategy.Deploy(from, to, desiredReplicas)
+	if err := strategy.Deploy(from, to, desiredReplicas); err != nil {
+		return err
+	}
+	return d.after(config, to)
```

Comment:
I think post hooks should also be able to cause deployments to fail.  I don't think our original arguments still hold - basically, if I want to fail a deployment because of a post hook, I should be allowed to.  Same for After.  I do think after needs to be able to specify the abort policy, but I don't think the current Post behavior is right.